Updating to have this encode as configured

goosemo opened the pull request:
Was getting some stack traces when this was writing out UTF-8 git commit log content. Made this change locally to address.

vaab commented:
Ok thx, can you tell us your platform (windows ? linux ? macosx ?, your python version ?)
I'll need to create a regression test on this bug to include your change. I guess it probably won't be as easy on all combination of platforms and version, but I can manage that.

goosemo commented:
So it was python2.7 on OSX.

I've just updated the code to be python version conditional, though that doesn't seem ideal. 
I also added in a test change to make catch this case for the future. When added, and w/o my changes I'm able to recreate the issue I saw:

```
======================================================================
ERROR: test_insertions (test.test_publish.FileInsertAtFirstRegexMatchTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mgoose/workspace/goosemo/gitchangelog/test/test_publish.py", line 184, in test_insertions
    self.assertEqual(make_insertion("A\nB\n", r"$", "✔\n", idx=lambda m: m.end() + 1), "A\nB\n✔\n")
  File "/Users/mgoose/workspace/goosemo/gitchangelog/test/test_publish.py", line 171, in make_insertion
    insert.splitlines(True))
  File "/Users/mgoose/workspace/goosemo/gitchangelog/src/gitchangelog/gitchangelog.py", line 1473, in _wrapped
    write_content(dst, content)
  File "/Users/mgoose/workspace/goosemo/gitchangelog/src/gitchangelog/gitchangelog.py", line 1449, in write_content
    f.write(content_line)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2714' in position 0: ordinal not in range(128)
```
